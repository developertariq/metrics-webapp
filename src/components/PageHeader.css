.page-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 100%;
  height: 180px;
  background-color: #ec4c8a;
  padding: 10px;
}

.page-image{
  max-width: 50%;
  height: 180px;
  opacity: 0.3;
  /* object-fit: cover; */
}

.page-title {
  max-width: 50%;
  align-items: flex-start;
  text-align: right;
}