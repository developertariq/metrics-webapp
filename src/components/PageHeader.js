import React from 'react';
import PropTypes from 'prop-types';
import './PageHeader.css';

const PageHeader = (props) => {
  const { map, name, population } = props;
  return (
    <div className="page-header">
      <img src={map} alt={name} className="page-image" />
      <div className="page-title">
        <h2 style={{ marginBottom: '0' }}>
          {name}
        </h2>
        {population}
      </div>
    </div>
  );
};
PageHeader.propTypes = {
  name: PropTypes.string,
  population: PropTypes.number,
  map: PropTypes.string,
};

PageHeader.defaultProps = {
  name: '',
  population: 0,
  map: '',
};

export default PageHeader;
