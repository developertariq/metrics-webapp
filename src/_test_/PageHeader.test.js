import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/configureStore';
import '@testing-library/jest-dom';
import PageHeader from '../components/PageHeader';

describe('PageHeader component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PageHeader />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
