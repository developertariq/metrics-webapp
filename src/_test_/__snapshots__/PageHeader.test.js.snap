// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component renders correctly 1`] = `
<div
  className="page-header"
>
  <img
    alt=""
    className="page-image"
    src=""
  />
  <div
    className="page-title"
  >
    <h2
      style={
        Object {
          "marginBottom": "0",
        }
      }
    />
    0
  </div>
</div>
`;
